src/curl.c

```diff
@@ -157,39 +157,58 @@ static size_t swupd_download_version_to_memory(void *ptr, size_t size, size_t nm
 /* curl easy CURLOPT_WRITEFUNCTION callback */
 size_t swupd_download_file(void *ptr, size_t size, size_t nmemb, void *userdata)
 {
-	struct file *file = (struct file *)userdata;
+	struct file *file = (struct file*)userdata;
 	const char *outfile;
 	int fd;
 	FILE *f;
-	size_t written;
+	size_t written, remaining;
 
 	outfile = file->staging;
+        if (file->fd_valid) {
+		fd = file->fd;
+        } else {
+		fd = open(outfile, O_CREAT | O_RDWR | O_CLOEXEC | O_APPEND, 00600);
+		if (fd < 0) {
+			LOG_ERROR(file, "Cannot open file for write", class_file_io,
+				  "\\*outfile=\"%s\",strerror=\"%s\"*\\", outfile, strerror(errno));
+			return -1;
+		}
+		file->fd = fd;
+		file->fd_valid = 1;
+        }
+
+        /* handle short writes with repeated write() calls */
+        for (remaining = size*nmemb; remaining; remaining -= written) {
```

Comment:
In soletta we provide a check-style script that calls uncrustify and if diff is non-0 we complain. Recommend doing so and using a semaphore-ci to ensure that on PR.